packages/legend-query-builder/src/stores/explorer/QueryBuilderExplorerState.ts

```diff
@@ -518,6 +521,29 @@ export const getQueryBuilderPropertyNodeData = (
     // returns we assign the type of the property node to the mapped subClass
     subClass,
   );
+
+  // Update parent's childrenIds for this proerty
```

Comment:
not really understanding the comments. 
seems we are removing a childId from the parent with the below code, so could you connect the comment to the actual code. 
Not sure I understand why we need to do this.

Reply:
The issue is that parent's child id doesn't match with this child's id.

this is the actual code itself....

use the example provided in this PR,

when generating the child id of `employees`, it calls `generateExplorerTreeClassNodeChildrenIDs(propertyNode)`,
the return value is employees.partyBase.

When visiting the child employees.partyBase, its id is employees.partyBase@my::Party, which is done in 

```
  subClass
      ? generateExplorerTreeSubtypeNodeID(
          generateExplorerTreePropertyNodeID(
            parentNode instanceof QueryBuilderExplorerTreeRootNodeData
              ? ''
              : parentNode.id,
            property.name,
          ),
          subClass.path,
        )
      : generateExplorerTreePropertyNodeID(
          parentNode instanceof QueryBuilderExplorerTreeRootNodeData
            ? ''
            : parentNode.id,
          property.name,
        )
```